### ISSUE TITLE
No error for minimum violation

### ISSUE THREAD

**Issue by F43nd1r** [sinclairzx81/typebox#277]
When I create an integer type with a minimum value, I'd expect to receive an error telling me about that. However, this:
```typescript
    const T = Type.Integer({ minimum: 0 })
    const Checker = TypeCompiler.Compile(T)
    console.log({
        check: Checker.Check(-1),
        errors: [...Checker.Errors(-1)],
    })
```
produces:
```
{ check: false, errors: [] }
```

Is this intended behaviour?

**Comment by sinclairzx81** [comment]
@F43nd1r Hi, thanks for reporting! Yeah, this appears to be a bug in the error reporter for some truth checks for potentially undefined numbers. Have pushed a fix for this on `0.25.6`.

Thanks for letting me know!
S

**Comment by sinclairzx81** [comment]
Have applied a couple more updates around this on the `Value.*` checking also, all fixes on `0.25.7`

Will leave the issue open for a day, but feel free to close if things check out ok your side.

Cheers!
S

**Comment by F43nd1r** [comment]
Hi, thanks for the fast reaction ❤️ 

Works well for us now.
